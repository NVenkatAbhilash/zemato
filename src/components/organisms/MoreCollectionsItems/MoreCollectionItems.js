import React from "react";
import { Grid } from "@material-ui/core";
import CollectionViewItem from "../../molecules/CollectionViewItem/CollectionViewItem";
import collection1 from "./../../../resources/images/collection1.jpg";
import collection2 from "./../../../resources/images/collection2.jpg";
import collection3 from "./../../../resources/images/collection3.jpg";
import collection4 from "./../../../resources/images/collection4.jpg";
import collection5 from "./../../../resources/images/collection5.jpg";
import { Link } from "react-router-dom";


const MoreCollectionItems = (props) => {

  return (
    <Grid container spacing={2} {...props}>
      {moreCollections.map((value, index) => {
        return (
          <Grid item md={3} sm={6} xs={12} key={index}>
            <Link
              to={{
                pathname: "/collections",
                state: {
                  foodImg: value.photo,
                  ...value,
                },
              }}
              style={{ textDecoration: "none" }}
            >
              <CollectionViewItem {...value} mode="small" />
            </Link>
          </Grid>
        );
      })}
    </Grid>
  );
};

const moreCollections = [
  {
    placesCount: 12,
    special: "New Year's Specials",
    foodImg: collection1,
  },
  {
    placesCount: 31,
    special: "Trending This Week",
    foodImg: collection2,
  },
  {
    placesCount: 111,
    special: "Hyderabad's Hottest",
    foodImg: collection3,
  },
  {
    placesCount: 221,
    special: "Work Friendly Places!",
    foodImg: collection4,
  },
  {
    placesCount: 12,
    special: "Live Sports Screenings",
    foodImg: collection5,
  },
  {
    placesCount: 6,
    special: "Newly Opened",
    foodImg: collection2,
  },
  {
    placesCount: 12,
    special: "Great Breakfasts",
    foodImg: collection4,
  },
  {
    placesCount: 8,
    special: "Happy Hours",
    foodImg: collection1,
  },
  {
    placesCount: 8,
    special: "Best Milkshakes",
    foodImg: collection3,
  },
  {
    placesCount: 221,
    special: "Work Friendly Places!",
    foodImg: collection4,
  },
  {
    placesCount: 12,
    special: "Live Sports Screenings",
    foodImg: collection5,
  },
  {
    placesCount: 6,
    special: "Newly Opened",
    foodImg: collection2,
  },
];

export default MoreCollectionItems;
