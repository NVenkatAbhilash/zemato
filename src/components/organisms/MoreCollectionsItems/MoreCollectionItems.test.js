import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import MoreCollectionItems from "./MoreCollectionItems";
import renderer from "react-test-renderer";

it("renders MoreCollectionItems snapshot correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MoreCollectionItems />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders MoreCollectionItems without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <MoreCollectionItems />
    </BrowserRouter>,
    div
  );
});
