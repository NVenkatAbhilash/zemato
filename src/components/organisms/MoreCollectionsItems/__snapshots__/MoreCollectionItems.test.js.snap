// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MoreCollectionItems snapshot correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection1.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            New Year's Specials
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            12
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection2.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Trending This Week
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            31
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection3.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Hyderabad's Hottest
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            111
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection4.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Work Friendly Places!
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            221
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection5.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Live Sports Screenings
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            12
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection2.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Newly Opened
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            6
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection4.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Great Breakfasts
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            12
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection1.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Happy Hours
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            8
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection3.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Best Milkshakes
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            8
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection4.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Work Friendly Places!
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            221
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection5.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Live Sports Screenings
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            12
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/collections"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-1 makeStyles-small-2 MuiGrid-container MuiGrid-align-items-xs-flex-end"
        data-testid="main"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection2.jpg)",
          }
        }
      >
        <div
          className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="special"
          >
            Newly Opened
          </h6>
          <h6
            className="MuiTypography-root MuiTypography-subtitle2"
            data-testid="place"
          >
            6
             Places
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 17l5-5-5-5v10z"
              />
            </svg>
          </h6>
        </div>
      </div>
    </a>
  </div>
</div>
`;
